### ISSUE TITLE
How to get Dark Mode to work in Qt apps in AppImages?

### ISSUE THREAD

**Issue by tpimh** [linuxdeploy/linuxdeploy-plugin-qt#155]
I am trying to package a Qt application in an AppImage. The app always has the default look regardless of the OS setting to use Dark Mode.

I tried to solve this problem by manually adding `libqt5ct.so` and `libqt5ct-style.so` to the AppImage, but this didn't change anything. My qt5ct outside of AppImage is set to use "Fusion" style with "darker" color scheme, but I still get the default look when launching the AppImage.

**Comment by srilakshmikanthanp** [comment]
I am facing the same issue have you managed to fix this?

**Comment by tpimh** [comment]
Nope, I believe this is just a limitation at the moment. I am not aware of anyone working on this.

**Comment by TheAssassin** [comment]
Someone will have to research what files need to be added exactly before this issue can be resolved.

**Comment by TheAssassin** [comment]
@srilakshmikanthanp does removing the hook altogether fix the problem, too? I'd consider removing it entirely (or at least not overwrite an existing value).

**Comment by TheAssassin** [comment]
Define users. End users of your application? Users of linuxdeploy who build app bundles?

**Comment by srilakshmikanthanp** [comment]
Users of linuxdeploy who build app bundles.

**Comment by TheAssassin** [comment]
As I said I am open to removing the hook altogether, I am not sure if it is still helpful at all. Plus, I already suggested that we can just accept a predefined value.

I am not sure that setting `xdgdesktopportal` is universally OK, especially on older, still-supported distros. That would have to be tested.

I am open to parameterizing this, though.

**Comment by scrubbbbs** [comment]
I also agree it would be better if this was not set at all, and let it be up to users running the AppImage. I used to use qt6ct on my desktop (which requires QT_QPA_PLATFORMTHEME=qt6ct) , and this hook would break that.

Also the gtk2 platform theme (`libqtgtk2.so`) is not typically installed in Qt6, and there is no apt package for it.

**Comment by TheAssassin** [comment]
I suppose we should get rid of the hook script for Qt 6 then at least. I'll try to add a switch later.

**Comment by TheAssassin** [comment]
The latest version no longer creates that hook file. Please test and report.

**Comment by dougmassay** [comment]
The default QT_QPA_PLATFORMTHEME=gtk3 was working in most of my use cases. My issue was that I was using a custom AppRun that was still adding the call to source the hook. So when the hook creation was removed 3 weeks ago, my Qt6 AppImage broke (when building with the latest linux-deploy-qt plugin). I tracked it down and just incorporated the contents of the hook script into my custom AppRun, but I was sure confused as to what had happened for a while. There was no link to this issue in the commit that removed the hook creation.

**Comment by bjorn** [comment]
> I tracked it down and just incorporated the contents of the hook script into my custom AppRun

Just wondering, but was that really necessary? Explicitly requesting the `gtk3` platform theme was causing issues with adjusting to dark mode (which are fixed by using `xdgdesktopportal` instead). Hence it was judged better to leave this variable alone and make it easier for the user to customize the platform theme when they need to.

**Comment by dougmassay** [comment]
I'm very much in "if it ain't broke, don't fix it mode" here. I'm not experiencing issues with changing themes (dark or light) on the fly on the DEs I've tested with when using the default hook that sets the QT_QPA_PLATFORMTHEME to gtk3 (on the DEs specified in the hook). My minimum is Ubuntu 22.04 so I don't typically have to worry about older DEs that might use gtk2. If I get any user reports of theme breakage, I'll of corse revisit. But for now, I'm satisfied with what the plugin's hook script was providing by default (prior to 3 weeks ago).

**Comment by bjorn** [comment]
@dougmassay I guess the related comments may be outdated because it looks like the `gtk3` platform theme learned to respect light and dark theme preference since Qt 6.8 (see https://github.com/qt/qtbase/commit/d0b4e8a601bc2f81ddb23c124acc99defa26c02f).

You may anyway want to try whether you can just remove it, because by the same "if it ain't broke, don't fix it mode" logic, you don't want to be setting the `QT_QPA_PLATFORMTHEME` unnecessarily. It will for sure mess with users trying to set it to something else for whatever reason.
